envoy-filters/mgw-source/filters/http/mgw-wasm-websocket/filter.cc

```diff
@@ -317,7 +317,7 @@ void MgwWebSocketContext::onLog() { LOG_TRACE(std::string("onLog " + std::to_str
 
 void MgwWebSocketContext::onDelete() {
   LOG_TRACE(std::string("onDelete " + std::to_string(id())));
-  this->stream_handler_->reset();
+  this->stream_handler_->close();
```

Comment:
Isn't reset is the better option. Once the mgwContext is deleted we have no purpose in keeping the stream half closed and let enforcer send whatever it is left. When this triggers, the websocket connection is no longer available.

Reply:
Without this change the error `io.grpc.StatusRuntimeException: CANCELLED: cancelled before receiving half close` keeps occurring for every successful websocket connections when they close. This is the case for all websocket tests we have in main branch.
- In [here](https://github.com/envoyproxy/envoy/blob/80d3c1750a18915f37fa65ca9d1bf3c26dd87d0b/test/extensions/filters/http/wasm/test_data/test_grpc_stream_cpp.cc#L51) they had only used `close()` which is why I thought it would be alright. 
- Also, according to the explanation [here](https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/blob/master/docs/wasm_filter.md#reset) reset() I get the feeling that reset is trying to close something before it is ready. I am not 100% sure about this but this change seems to stop the error.

Reply:
I have tested keeping close and enabling trace logs for wasm filter. It seems if the request is half close, the server is not closing the connection (it seems grpc library inside enforcer is also outdated a bit). Hence it causes the stream_handlers within wasm filter to accumulate which could lead to memory leak.

Reply:
Can this accumilation be due to not calling `responseStreamObserver.onCompleted()` in https://github.com/wso2/product-microgateway/blob/2b8672d4d00d6fed01251579855d50d5ecec2095/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/websocket/WebSocketResponseObserver.java#L135 ?

Reply:
But we can't call it explicitly in onNext method right? Because termination is always initiated from the client. (wasm filter)

Reply:
Tested with the directions on switching-on trace and looking for the log `handler destructed`. The grpc handler instance did not get destructed as you mentioned. After trying out the above and a few other ways, just found out that all we had to do was call both `close()` and `reset()`. Updated the PR accordingly.

Reply:
great stuff. Reset after a half close does not violate the protocol it seems too.
+1